eclipse-jdt/eclipse.jdt.ui
# Fix buggy Javadoc styling enhancements menu +hard to read colored type parameters (#1442)

## What it does
Fixes buggy styling enhancements menu (see https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/1442#issuecomment-2154709945):
- menu content not properly reflecting enable/disable state of styling enhancements
- menu visually corrupted on Linux / GTK in certain situations (eg. after hover's shell is activated)

Also makes Javadoc to render (generic) type parameters with same font-weight (was lighter) like the rest of the method signature - see https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/1442#issuecomment-2155658440

## How to test
Toggle / change all styling enhancements settings via styling enhancements menu on JavadocHover's and JavadocView's tollbar and confirm styling enhancements menu reflect current settings + javadoc content being displayed and available actions have expected results (eg. toggling  / changing / resetting colors).
Scenarios of using styling enhancements menu to consider:
- in JavadocHover that was activated = by pressing F2, clicking inside content (including on toolbar after entering into with mouse, or on any tollbar item button (but not button's sub-menu arrow))
- in JavadocHover that was not activated = just hovered into by mouse, then opening enhancements menu by clicking button's sub-menu arrow
- in JavadocView that is displaying javadoc for some source code element
- in JavadocView that is not displaying any javadoc = view was just opened and no source code element was focused on yet
- current settings presented in enhancements menus in JavadocHover and JavadocView should be in sync

## Author checklist
